self_hosting_machinery/finetune/scripts/auxiliary/file_sets_context.py

```diff
@@ -37,6 +39,29 @@ def get_loss_by_content(self, model_name: str, content: str) -> Optional[float]:
         h = hashlib.sha1(content.encode("utf-8")).hexdigest()
         return self.loss_per_hash_db[(h, model_name)]["loss"] if (h, model_name) in self.loss_per_hash_db else None
 
+    def is_up_to_date(self) -> bool:
```

Comment:
get back `is_up_to_date` to prevent long finetune start when the files are already filtered